File: solutions/csharp/clock/1/Clock.cs
using System;
public class Clock : IEquatable<Clock>
{
    private int totalMinutes;
    public Clock(int hours, int minutes) => totalMinutes = hours * 60 + minutes >= 0 ? (hours % 24) * 60 + minutes % 1440 : (hours * 60 + minutes) % 1440 + 1440;
    public Clock Add(int minutesToAdd) => new Clock(totalMinutes / 60, totalMinutes % 60 + minutesToAdd);
    public Clock Subtract(int minutesToSubtract) => new Clock(totalMinutes / 60, totalMinutes % 60 - minutesToSubtract);
    public override string ToString() => $"{(totalMinutes / 60) % 24:D2}:{totalMinutes % 60:D2}";
    public bool Equals(Clock other) => totalMinutes == other.totalMinutes;
}

File: solutions/javascript/clock/1/clock.js
//
// This is only a SKELETON file for the 'Clock' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Clock {
  constructor(hours = 0, minutes = 0) {
    hours += Math.floor(minutes / 60)
    hours = hours % 24
    minutes  = minutes % 60
    this.hours = (hours < 0) ? hours + 24 : hours
    this.minutes  = (minutes < 0) ? minutes + 60 : minutes
  }

  toString() {
    return `${Clock._pad(this.hours)}:${Clock._pad(this.minutes)}`
  }

  plus(m = 0) {
    return new Clock(this.hours, this.minutes + m)
  }

  minus(m = 0) {
    return new Clock(this.hours, this.minutes - m)
  }

  equals(clock) {
    return this.toString() === clock.toString()
  }

  static _pad = (num) => num < 10 ? '0' + num : num.toString()
}
